src/literal.h

```diff
@@ -759,6 +759,7 @@ template<> struct hash<wasm::Literal> {
         case wasm::Type::exnref:
         case wasm::Type::anyref:
         case wasm::Type::eqref:
+        case wasm::Type::dataref:
```

Comment:
I think you're right. Any pass that uses hashing on GC atm looks like it would assert there (before this PR too) so we'll need to fix that eventually.